spec/lib/govuk_proxy/static_proxy_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "spec_helper"
+require "govuk_app_config/govuk_proxy/static_proxy"
+require "rack/mock"
+
+RSpec.describe GovukProxy::StaticProxy do
+  # dummy app to validate success
+  let(:app) { ->(_env) { [200, {}, "success"] } }
+  let(:app_domain) { "app.domain" }
+  let(:static_domain) { "static.domain" }
+
+  subject { GovukProxy::StaticProxy.new(app, backend: "https://static.domain", streaming: false) }
+
+  it "redirects the request if path begins with /asset/static" do
+    test_request("/assets/static/a.css", true)
+  end
+
+  it "ignores requests not with path prefix /asset/static" do
+    test_request("/assets/app/a.css", false)
+  end
+
+  it "ignores requests where /asset/static isn't a prefix" do
+    test_request("/another/prefix/assets/static/a.css", false)
+  end
+
+  it "ignores requests with no path" do
+    test_request("/", false)
+  end
+end
+
+def test_request(path, proxied)
+  dest_host = (proxied ? static_domain : app_domain)
+
+  stub_request(:get, "https://#{dest_host}#{path}")
+      .with(headers: { "Host" => dest_host, "X-Forwarded-For" => "" })
```

Comment:
Would it also be useful to simulate `X-Forwarded-Host` here? (I don't know if it matters, just something to consider)

Reply:
So don't really need X-Forwarded-Host here - we aren't doing anything with it the proxy, nor do we need it for any of the request we forward onto Static for development. 

Accidentally left in copied from a template. 🤦🏻